core/sdk-core/src/main/java/software/amazon/awssdk/core/async/AsyncResponseTransformer.java

```diff
@@ -260,7 +286,71 @@ static <ResponseT extends SdkResponse> AsyncResponseTransformer<ResponseT, Respo
      * </pre>
      */
     static <ResponseT extends SdkResponse>
-            AsyncResponseTransformer<ResponseT, ResponseInputStream<ResponseT>> toBlockingInputStream() {
+        AsyncResponseTransformer<ResponseT, ResponseInputStream<ResponseT>> toBlockingInputStream() {
         return new InputStreamResponseTransformer<>();
     }
+
+    /**
+     * Helper interface containing the result of {@link AsyncResponseTransformer#split(SplittingTransformerConfiguration)
+     * splitting} an AsyncResponseTransformer. This class holds both the publisher of the individual
+     * {@code AsyncResponseTransformer<ResponseT, ResponseT>} and the {@code CompletableFuture <ResultT>} which will
+     * complete when the {@code AsyncResponseTransformer} that was split itself would complete.
+     *
+     * @param <ResponseT> ResponseT of the original AsyncResponseTransformer that was split.
+     * @param <ResultT>   ResultT of the original AsyncResponseTransformer that was split.
+     * @see AsyncResponseTransformer#split(SplittingTransformerConfiguration)
+     */
+    interface SplitResult<ResponseT, ResultT>
```

Comment:
Can we move it outside of this interface? It's a bit hard to read

Reply:
We decided during design review to have it as a inner interface

Reply:
Is the reason that's we want it to be under `AsyncResponseTransformer`, as in `AsyncResponseTransformer.SplitResult`?

Reply:
Yes, we had an extensive discussion on naming this class, and we ended up choosing it to be an inner class as `AsyncResponseTransformer.SplitResult` which seemed to be the most clear option at the time